packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -918,16 +918,18 @@ class RenderParagraph extends RenderBox
       );
 
       if (info.isPlaceholder) {
-        final SemanticsNode childNode = children.elementAt(placeholderIndex++);
-        final TextParentData parentData = child!.parentData as TextParentData;
-        childNode.rect = Rect.fromLTWH(
-          childNode.rect.left,
-          childNode.rect.top,
-          childNode.rect.width * parentData.scale!,
-          childNode.rect.height * parentData.scale!,
-        );
-        newChildren.add(childNode);
-        child = childAfter(child);
+        if (children.isNotEmpty) {
+          final SemanticsNode childNode = children.elementAt(placeholderIndex++);
```

Comment:
If children is empty (seen in the example app posted in the bug and in the test case, this throws an exception)